pkg/hubble/filters/workload_test.go

```diff
@@ -0,0 +1,630 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Hubble
+
+//go:build !privileged_tests
+
+package filters
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	flowpb "github.com/cilium/cilium/api/v1/flow"
+	v1 "github.com/cilium/cilium/pkg/hubble/api/v1"
+)
+
+func TestWorkloadFilterInclude(t *testing.T) {
+	type args struct {
+		f  []*flowpb.FlowFilter
+		ev *v1.Event
+	}
+	tests := []struct {
+		name string
+		args args
+		want bool
+	}{
+		{
+			name: "source-nil",
+			args: args{
+				f: []*flowpb.FlowFilter{
+					{
+						SourceWorkload: []*flowpb.Workload{
+							{
+								Kind: "Deployment",
+								Name: "hubble-relay",
+							},
+						},
+					},
+				},
+				ev: nil,
+			},
+			want: false,
+		},
+		{
+			name: "destination-nil",
+			args: args{
+				f: []*flowpb.FlowFilter{
+					{
+						DestinationWorkload: []*flowpb.Workload{
+							{
+								Kind: "Deployment",
+								Name: "hubble-relay",
+							},
+						},
+					},
+				},
+				ev: nil,
+			},
+			want: false,
+		},
+		{
+			name: "source-filter-nil",
+			args: args{
+				f: []*flowpb.FlowFilter{
+					{
+						SourceWorkload: nil,
+					},
+				},
+				ev: &v1.Event{Event: &flowpb.Flow{
+					Source: &flowpb.Endpoint{Workloads: []*flowpb.Workload{
+						{
+							Kind: "Deployment",
+							Name: "hubble-relay",
+						},
+					}},
+				}},
+			},
+			want: true,
+		},
+		{
+			name: "destination-filter-nil",
+			args: args{
+				f: []*flowpb.FlowFilter{
+					{
+						DestinationWorkload: nil,
+					},
+				},
+				ev: &v1.Event{Event: &flowpb.Flow{
+					Source: &flowpb.Endpoint{Workloads: []*flowpb.Workload{
```

Comment:
Is it intended that the source is populated here? I expect the `nil` filter to match anything, so this is not a correctness issue, just a bit weird.

Reply:
Good catch this is definitely a typo, should be fixed now :pray: